packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -272,6 +114,95 @@ class InputDateTimePicker extends React.Component {
 		this.componentContainerEvents.push(e);
 	}
 
+	/** *******************************************************************************************
+	 * Component logic
+	 *********************************************************************************************/
+	componentWillReceiveProps(nextProps) {
+		const newSelectedDateTime = nextProps.selectedDateTime;
+
+		const needDateTimeStateUpdate =
+			newSelectedDateTime !== this.props.selectedDateTime && // selectedDateTime props updated
+			newSelectedDateTime !== this.state.datetime && // not the same ref as state date time
+			!isSameMinute(newSelectedDateTime, this.state.datetime); // not the same value as state
+
+		if (needDateTimeStateUpdate) {
+			const dateRelatedPartState = extractDateTimeParts(newSelectedDateTime);
+			this.setState(dateRelatedPartState);
+		}
+	}
+
+	onChangeInput(event) {
+		const textInput = event.target.value;
+		let nextState;
```

Comment:
Useless variable.
Should be a single if with two return.

Reply:
In fact the return is the `this.onChange` call.
It takes the next state as parameter. Do you mean you prefer to return `this.onChange` 2 times  (I mean 1 per branch) instead of this local var that is populated differently in the if/else ?

Reply:
Done